modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItRaftGroupServiceTest.java

```diff
@@ -0,0 +1,155 @@
+import org.apache.ignite.raft.client.service.RaftGroupService;
+import org.apache.ignite.utils.ClusterServiceTestUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * Integration test methods of raft group service.
+ */
+@ExtendWith(WorkDirectoryExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ItRaftGroupServiceTest {
+    @WorkDirectory
+    private static Path workDir;
+
+    private static final int NODES_CNT = 2;
+
+    private static final int NODE_PORT_BASE = 20_000;
+
+    private static final TestScaleCubeClusterServiceFactory NETWORK_FACTORY = new TestScaleCubeClusterServiceFactory();
+
+    private static final String RAFT_GROUP_NAME = "part1";
+
+    private static List<ClusterService> clusterServices = new ArrayList<>();
+
+    private static List<Loza> raftSrvs = new ArrayList<>();
+
+    private static Map<ClusterNode, CompletableFuture<RaftGroupService>> raftGroups = new HashMap<>();
+
+    @BeforeAll
+    public static void beforeAll(TestInfo testInfo) throws Exception {
+        List<NetworkAddress> localAddresses = findLocalAddresses(NODE_PORT_BASE,
+                NODE_PORT_BASE + NODES_CNT);
+
+        var nodeFinder = new StaticNodeFinder(localAddresses);
+
+        for (int i = 0; i < NODES_CNT; i++) {
+            ClusterService clusterService = ClusterServiceTestUtils.clusterService(
+                    testInfo,
+                    NODE_PORT_BASE + i,
+                    nodeFinder,
+                    NETWORK_FACTORY
+            );
+
+            clusterServices.add(clusterService);
+
+            clusterService.start();
+        }
+
+        assertTrue(waitForTopology(clusterServices.get(NODES_CNT - 1), NODES_CNT, 1000));
+
+        List<ClusterNode> nodes = clusterServices.stream().map(cs -> cs.topologyService().localMember()).collect(Collectors.toList());
+
+        for (int i = 0; i < NODES_CNT; i++) {
+            Loza raftServer = new Loza(clusterServices.get(i), workDir);
+
+            raftSrvs.add(raftServer);
+
+            raftServer.start();
+
+            CompletableFuture<RaftGroupService> raftGroupServiceFuture = raftSrvs.get(i).prepareRaftGroup(
+                    RAFT_GROUP_NAME,
+                    nodes,
+                    () -> mock(RaftGroupListener.class)
+            );
+
+            raftGroups.put(clusterServices.get(i).topologyService().localMember(), raftGroupServiceFuture);
+        }
+
+        raftGroups.get(clusterServices.get(0).topologyService().localMember()).get();
+        raftGroups.get(clusterServices.get(1).topologyService().localMember()).get();
+    }
+
+    @AfterAll
+    public static void afterAll() throws Exception {
+        for (Loza raftSrv : raftSrvs) {
```

Comment:
We should shutdown clients from `raftGroups`, shouldn't we? I see, that `org.apache.ignite.raft.jraft.rpc.impl.RaftGroupServiceImpl#shutdown` is empty now, but it can be defined better in future.

Reply:
Fixed.